okhttp-client/src/main/scala/org/http4s/client/okhttp/OkHttp.scala

```diff
@@ -0,0 +1,140 @@
+package org.http4s.client.okhttp
+
+import java.io.IOException
+
+import cats.data._
+import cats.effect._
+import cats.effect.implicits._
+import okhttp3.{
+  Call,
+  Callback,
+  OkHttpClient,
+  Protocol,
+  RequestBody,
+  Headers => OKHeaders,
+  MediaType => OKMediaType,
+  Request => OKRequest,
+  Response => OKResponse
+}
+import okio.BufferedSink
+import org.http4s.{Header, Headers, HttpVersion, Method, Request, Response, Status}
+import org.http4s.client.{Client, DisposableResponse}
+import fs2.Stream._
+import fs2._
+import fs2.io._
+
+import scala.collection.JavaConverters._
+import scala.concurrent.ExecutionContext
+
+object OkHttp {
+
+  def defaultconfig[F[_]]()(implicit F: Effect[F]): F[OkHttpClient.Builder] = F.delay {
+    new OkHttpClient.Builder()
+      .protocols(
+        List(
+          Protocol.HTTP_2,
+          Protocol.HTTP_1_1
+        ).asJava)
+  }
+
+
+  def apply[F[_]](config: F[OkHttpClient.Builder] = null)(
+      implicit F: Effect[F],
+      ec: ExecutionContext): F[Client[F]] = F.map(Option(config).getOrElse(defaultconfig[F]())) { c =>
+    val client = c.build()
+    Client(
+      Kleisli { req =>
+        F.async[DisposableResponse[F]] { cb =>
+          client.newCall(toOkHttpRequest(req)).enqueue(handler(cb))
+          ()
+        }
+      },
+      F.delay({
+        client.dispatcher.executorService().shutdown()
+        client.connectionPool().evictAll()
+        if (client.cache() != null) {
+          client.cache().close()
+        }
```

Comment:
Can any of these operations fail?  I'm surprised that there isn't a single cleanup method on OkHttp that handles this for you.

Reply:
https://square.github.io/okhttp/3.x/okhttp/okhttp3/OkHttpClient.html

the `Shutdown isn't necessary` section describes shutting it down this way, I should probably wrap them in case they fail

Reply:
> The threads and connections that are held will be released automatically if they remain idle.

This seems like it would depend on the type of executor passed in?

That section makes me feel better that it's not a crisis if any of these steps fail, but yeah, wrapping for best effort is probably wise.